ambry-commons/src/test/java/com.github.ambry.commons/ResponseHandlerTest.java

```diff
@@ -106,29 +108,67 @@ public ReplicaId getLastReplicaID() {
+    handler.onEvent(new MockReplicaId(), new ConnectionPoolTimeoutException(""));
+    Assert.assertEquals(1, map.getLastReplicaEvents().size());
     Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Node_Timeout));
+
     map.reset();
-    handler.onRequestResponseError(new MockReplicaId(), ServerErrorCode.IO_Error);
+    handler.onEvent(new MockReplicaId(), ServerErrorCode.IO_Error);
+    Assert.assertEquals(2, map.getLastReplicaEvents().size());
     Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Node_Response));
     Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Disk_Error));
+
     map.reset();
-    handler.onRequestResponseError(new MockReplicaId(), ServerErrorCode.Disk_Unavailable);
+    handler.onEvent(new MockReplicaId(), ServerErrorCode.Disk_Unavailable);
+    Assert.assertEquals(2, map.getLastReplicaEvents().size());
     Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Node_Response));
     Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Disk_Error));
+
+    map.reset();
+    handler.onEvent(new MockReplicaId(), ServerErrorCode.Partition_ReadOnly);
+    Assert.assertEquals(3, map.getLastReplicaEvents().size());
+    Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Partition_ReadOnly));
+    Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Node_Response));
+    Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Disk_Ok));
+
     map.reset();
-    handler.onRequestResponseError(new MockReplicaId(), ServerErrorCode.Partition_ReadOnly);
+    handler.onEvent(new MockReplicaId(), ServerErrorCode.Unknown_Error);
+    Assert.assertEquals(2, map.getLastReplicaEvents().size());
     Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Node_Response));
     Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Disk_Ok));
+
     map.reset();
-    handler.onRequestResponseError(new MockReplicaId(), ServerErrorCode.Unknown_Error);
+    handler.onEvent(new MockReplicaId(), ServerErrorCode.No_Error);
+    Assert.assertEquals(2, map.getLastReplicaEvents().size());
     Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Node_Response));
     Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Disk_Ok));
+
+    map.reset();
+    handler.onEvent(new MockReplicaId(), NetworkClientErrorCode.NetworkError);
+    Assert.assertEquals(1, map.getLastReplicaEvents().size());
+    Assert.assertTrue(map.getLastReplicaEvents().contains(ReplicaEventType.Node_Timeout));
+
+    map.reset();
+    handler.onEvent(new MockReplicaId(), NetworkClientErrorCode.ConnectionUnavailable);
+    Assert.assertEquals("Unrecognized events should be ignored", 0, map.getLastReplicaEvents().size());
+
+    map.reset();
+    handler.onEvent(new MockReplicaId(), new RouterException("", RouterErrorCode.UnexpectedInternalError));
```

Comment:
Nope. Just verifying that an unrecognized event is simply ignored, and does not cause any exceptions or other effects. Callers should be able to pass in anything.